src/include/sof/audio/pipeline.h

```diff
@@ -129,6 +129,10 @@ struct pipeline_task {
 	struct task task;		/**< parent structure */
 	bool registrable;		/**< should task be registered on irq */
 	struct comp_dev *sched_comp;	/**< pipeline scheduling component */
+	/* private data used by Zephyr DMA domain for internal
+	 * housekeeping - don't touch.
+	 */
+	void *priv;
```

Comment:
I wouldn't mention about Zephyr DMA domain here. Other's might use this also in the future. This is common enough inside structures so a generic comment like `private data for internal housekeeping` would be enough.

Reply:
cannot you just add your `ctrl_data` pointer to `struct zephyr_ll_pdata` stored in `task.priv_data`?